docs/application/native/guides/device/diagnostics.md

```diff
@@ -51,54 +51,57 @@ To enable your application to use Diagnostics API:
    #include <diagnostics.h>
    ```
 
-## Crash reports
+## Diagnostic events
 
-Diagnostics allows you to read the crash reports in case the process in the system fails. To do so, you must register the callback that will be called when the crash report is created:
+Diagnotics API allow you to send and receive diagnostic events. The events can contain additional data packed into bundle.
+To send diagnostic event, use `diagnostics_send_event()`:
 
 ```cpp
-void crash_callback(diagnostics_ctx_h ctx, void *user_data)
+int main(int argc, char *argv[])
 {
-    ...
-}
+    bundle *event_data = bundle_create();
 
-int register_callback(void *user_data)
-{
-    return diagnostics_set_notification_cb(crash_callback, user_data);
+    // This is required for system services only, not applications
+    diagnostics_set_client_id("diagnostics-client");
+
+    bundle_add_str(event_data, "somekey", "somevalue");
+    diagnostics_send_event("ConnectionBroken", event_data);
+
+    bundle_free(event_data);
 }
 ```
 
-To deregister the callback, use `diagnostics_unset_notification_cb()`.
-
-In case a new crash report is created, callback is called. To read the content of the crash report use `diagnostics_get_data()`:
+To receive diagnostic evnet, use `diagnostics_subscribe_event()` proceded by `diagnostics_set_notification_cb()`:
 
 ```cpp
-void crash_callback(diagnostics_ctx_h ctx, void *user_data)
+void notification_handler(diagnostics_ctx_h ctx, void *user_data)
 {
-    diagnostics_data_h data;
-    const char *params[] = {"cs_info_json"};
+    ...
 
-    int ret = diagnostics_get_data(ctx, params, 1, &data);
-    if (ret != DIAGNOSTICS_ERROR_NONE)
-        return;
+    diagnostics_destroy(ctx);
+}
 
-    ...
+int main(int argc, char **argv)
+{
+    diagnostics_set_notification_cb(notification_handler, NULL);
+    diagnostics_subscribe_event("ConnectionBroken", "org.tizen.someapp");
 }
 ```
 
-Use the `params` argument to specify which data to read from the crash report. Currently, the proper values are:
+## Diagnostic data request
 
--   `"cs_info_json"`: JSON report is returned, which contains information about crashed process such as callstack, mapped memory regions, and so on.
--   `"cs_full"`: Full crash report ZIP archive is collected.
+Diagnostics API also can request data from other application or service, especially the diagnostic data for various reasons cannot or must not be saved in the logs, such as binary or in custom format data, large volume data, reports that are time-consuming to create, and so on.
```

Comment:
This is a bit confusing:
especially the diagnostic data for various reasons cannot or must not be saved in the logs, such as binary or in custom format data, large volume data, reports that are time-consuming to create, and so on.

such as binary or in custom format data it cannot be saved? or it can be requested ?

Reply:
The point is that with diagnostics API it is possible to request for data that has not been logged anywhere so far because its format did not allow it (binary data, large amounts of data, etc.).
Now any data (including but not limited to binary) can be retrieved using diagnostics API

Reply:
@mmoscicki2, would this be better:

Diagnostics API also can request data from other applications or services, especially you can request the diagnostic data that could not be saved in the logs due to various reasons. For example, if the volume of data is too large, the data format is not supported as it is in binary or custom format, reports that are time-consuming to create, and so on, such data can now be requested using this API.